src/librustdoc/html/ambiguity.rs

```diff
@@ -0,0 +1,202 @@
+use rustc_data_structures::fx::FxHashMap;
```

Comment:
Might be worth adding a comment explaining how the path ambiguities are handled at the top of the file.